glass/user.py

```diff
@@ -45,3 +47,52 @@ def load_cls(filename):
         values = npz['values']
         split = npz['split']
     return np.split(values, split)
+
+
+class _FitsWriter:
+    '''Writer that creates a FITS file.  Initialised with the fits object and extention name.'''
+
+    def __init__(self, fits, ext=None):
+        '''Create a new, uninitialised writer.'''
+        self.fits = fits
+        self.ext = ext
+
+    def _append(self, data, names=None):
+        '''Internal method where the FITS writing is done'''
+
+        if self.ext is None or self.ext not in self.fits:
+            self.fits.write_table(data, names=names, extname=self.ext)
+            if self.ext is None:
+                self.ext = self.fits[-1].get_extnum()
+        else:
+            hdu = self.fits[self.ext]
+            # not using hdu.append here because of incompatibilities
+            hdu.write(data, names=names, firstrow=hdu.get_nrows())
+
+    def write(self, data=None, /, **columns):
+        '''Writes to FITS by calling the internal _append method.
+        Pass either a positional variable (data)
+        or multiple named arguments (**columns)'''
+
+        # if data is given, write it as it is
+        if data is not None:
+            self._append(data)
+
+        # if keyword arguments are given, treat them as names and columns
+        if columns:
+            names, values = list(columns.keys()), list(columns.values())
+            self._append(values, names)
+
+
+@contextmanager
+def write_catalog(filename, *, ext=None):
+    '''Context manager for a FITS catalogue writer.  Calls class FitsWriter.
+
+    ext is the name of the HDU extension
+
+    '''
```

Comment:
I was going to merge this, but then I saw this in the rendered docs. Could we create a slightly more user-friendly docstring?

```suggestion
    """
    Write a catalogue into a FITS file, where *ext* is the optional
    name of the extension.  To be used as a context manager::

        # create the catalogue writer
        with write_catalog("catalog.fits") as out:
            ...
            # write catalogue columns RA, DEC, E1, E2, WHT with given arrays
            out.write(RA=lon, DEC=lat, E1=eps1, E2=e2, WHT=w)

    """
```